File: JAVA-BASIC/src/main/java/_generics/DogsAndRobots.py


class Dog:

    def speak(self):
        print("Arf!")

    def sit(self):
        print("Sitting!")

    def reproduce(self):
        pass

class Robot:
    def speak(self):
        print("click!")
    def sit(self):
        print("Clank!")
    def oilChange(self):
        pass

def perform(anything):
    anything.speak()
    anything.sit()

if __name__ == '__main__':
    a = Dog()
    b = Robot()
    perform(a)
    perform(b)

File: java/JAVA-BASIC/src/main/java/_onjava8/_generics/DogsAndRobots.cpp
##include <iostream>

using namespace std;


class Dog {
public:
    void speak() { cout << "Arf!" << endl; }
    void siet() { count << "Sitting" << endl; }
    void reproduce() {}
}

class Robot {
public:
    void speak() { cout << "Click!" << endl; }
    void sit() { count << "Clank!" << endl; }
    void oilChange() {}
}

template<class T> void perform(T anything) {
    anything.speak()
    anything.sit()
}

int main() {
    Dog d;
    Robot r;
    perform(d);
    perform(r);
}